portfolio/src/main/java/com/google/sps/servlets/DiscussionServlet.java

```diff
@@ -1,47 +1,72 @@
-
 package com.google.sps.servlets;
-
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
 import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/discussion")
 public class DiscussionServlet extends HttpServlet {
 
-  private ArrayList<String> classicRock;
-  
   @Override
-  public void init(){
-    
-    classicRock = new ArrayList<>();
-    
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+
+    ArrayList<String> classicRock = new ArrayList<>();
     classicRock.add("Sultans of Swing");
     classicRock.add("Thunderstruck");
     classicRock.add("Burnin' for you");
     classicRock.add("Holy Diver");
 
+    Query query = new Query("Classic-Rock").addSort("commentTime", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery result = datastore.prepare(query);
+
+    for (Entity entity : result.asIterable()) {
+      String songName = (String) entity.getProperty("songName");
+      classicRock.add(songName);
+    }
+
+    String json = convertToJson(classicRock);
+    response.getWriter().println(json);
   }
 
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("application/json;");
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String songName = getParameter(request, "song-name", "");
 
-    String json = convertToJson(classicRock);
+    Entity song = new Entity("Classic-Rock");
+    song.setProperty("songName", songName);
+    song.setProperty("commentTime", System.currentTimeMillis());
```

Comment:
While this works, is there a better way that you know to save the ordering rather than just saving the timestamp?

Reply:
I like timestamp because it gives you an easy way to show the most relevant or oldest messages. If you have particular songs you want to always show, perhaps include a "rank" field and use integers to rank their importance. Then, based on the integer assigned to each song you can display the ones you desire.

Reply:
Agreed, I think timestamp makes sense to use here